sympy/solvers/decompogen.py

```diff
@@ -42,7 +43,7 @@ def decompogen(f, symbol):
 
     # ===== Convert to Polynomial ===== #
     fp = Poly(f)
-    gens = fp.gens
+    gens = list(filter(lambda x: symbol in x.atoms(Symbol) , fp.gens))
```

Comment:
You should almost always be using something like `symbol in x.free_symbols` because `symbol` might appear in `x` as a dummy variable, e.g. `Integral(x, (x, 1, 2))` is independent of `x` -- it's not an x-dependent generator (if it appeared as a generator).